### PR TITLE
Responsive default size

### PR DESCRIPTION
**Pull request by FedeClaudi**
Building on @t-bltg 's work on `DEFAULT_WIDTH`, I've turned it into a function and added a second default width for error stacktraces. The motivation for this is that if the terminal gets resized or a different `IO` is used, `default_width` is updated accordingly. The idea is to use the deafult width of 88 when possible, but smaller ones if the `IO` is narrower to avoid messes like:
image

Now for instance stack trace panels will have a width of 120 when possible, but will be narrower for narrow terminals (the first stacktraces was done with a narrow terminal):
image

### CODE REVIEW

**Comment by FedeClaudi** [comment]
Disabled some testing pending: https://github.com/FedeClaudi/Term.jl/pull/114

**Comment by FedeClaudi** [comment]
Boy does this make testing even more annoying...

Todo: setup tests to use `STDOUT`-like stuff with known size:
1. avoid tests passing locally but failing on GH
2. test across multiple terminal sizes

**Comment by FedeClaudi** [comment]
Had to disable some `Panel` tests despite https://github.com/FedeClaudi/Term.jl/pull/114 because width of stuff varies - will need to fix how tests are done.

File: test/08_test_panel.jl [hunk 1]
```diff
@@ -303,19 +303,19 @@ id est laborum.""",
     @test size(p.measure) == (22, 49)
 end
 
-@testset "\e[34mPanel + renderables" begin
-    @testpanel(Panel(RenderableText("x"^5)), 3, 11)
+# @testset "\e[34mPanel + renderables" begin
+#     @testpanel(Panel(RenderableText("x"^5)), 3, 11)
 
-    @testpanel(Panel(RenderableText("x"^500); fit = false), 9, default_width())
+#     @testpanel(Panel(RenderableText("x"^500); fit = false), 9, default_width())
```
Comment (t-bltg) [inline test/08_test_panel.jl:324]: I think you can restore these tests and use `h = nothing` in `@testpanel`

Comment (FedeClaudi) [reply test/08_test_panel.jl:324]: good call

**Pull request merged**